AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_materialproviders/references.py

```diff
@@ -0,0 +1,652 @@
+# Code generated by smithy-python-codegen DO NOT EDIT.
+
+import Wrappers
+import abc
+import asyncio
+import aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders.dafny_to_smithy
+from aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders.deserialize import (
+    _deserialize_error as aws_cryptography_materialproviders_deserialize_error,
+)
+import aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders.models
+import aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders.smithy_to_dafny
+import botocore.client
+import module_
+import software_amazon_cryptography_materialproviders_internaldafny_types
+from software_amazon_cryptography_materialproviders_internaldafny_types import (
+    DecryptMaterialsInput_DecryptMaterialsInput as DafnyDecryptMaterialsInput,
+    DecryptMaterialsOutput_DecryptMaterialsOutput as DafnyDecryptMaterialsOutput,
+    DeleteCacheEntryInput_DeleteCacheEntryInput as DafnyDeleteCacheEntryInput,
+    Error_AwsCryptographicMaterialProvidersException,
+    GetBranchKeyIdInput_GetBranchKeyIdInput as DafnyGetBranchKeyIdInput,
+    GetBranchKeyIdOutput_GetBranchKeyIdOutput as DafnyGetBranchKeyIdOutput,
+    GetCacheEntryInput_GetCacheEntryInput as DafnyGetCacheEntryInput,
+    GetCacheEntryOutput_GetCacheEntryOutput as DafnyGetCacheEntryOutput,
+    GetClientInput_GetClientInput as DafnyGetClientInput,
+    GetEncryptionMaterialsInput_GetEncryptionMaterialsInput as DafnyGetEncryptionMaterialsInput,
+    GetEncryptionMaterialsOutput_GetEncryptionMaterialsOutput as DafnyGetEncryptionMaterialsOutput,
+    OnDecryptInput_OnDecryptInput as DafnyOnDecryptInput,
+    OnDecryptOutput_OnDecryptOutput as DafnyOnDecryptOutput,
+    OnEncryptInput_OnEncryptInput as DafnyOnEncryptInput,
+    OnEncryptOutput_OnEncryptOutput as DafnyOnEncryptOutput,
+    PutCacheEntryInput_PutCacheEntryInput as DafnyPutCacheEntryInput,
+    UpdateUsageMetadataInput_UpdateUsageMetadataInput as DafnyUpdateUsageMetadataInput,
+)
+from typing import Any, Dict
+
+
+class IBranchKeyIdSupplier(metaclass=abc.ABCMeta):
```

Comment:
What is ABCMeta?

Reply:
This module: https://docs.python.org/3/library/abc.html
"abc" -> "abstract base class". It's a Pythonic way to write interfaces.